infrastructure/smart-contracts/contracts/drops/TinyKeysAirdrop.sol

```diff
@@ -104,66 +108,72 @@ contract TinyKeysAirdrop is Initializable, AccessControlUpgradeable {
-        // If the airdrop counter exceeds the total supply at start this means the airdrop is complete
-        airdropCounter = endingKeyId < totalSupplyAtStart ? endingKeyId + 1 : endingKeyId;
+        airdropCounter = endingKeyId + 1;
         emit AirdropSegmentComplete(startingKeyId, endingKeyId);
     }
 
-    function processAirdropSegmentOnlyStake(uint256[] memory keyIds) external onlyRole(DEFAULT_ADMIN_ROLE)  {
+    function processAirdropSegmentOnlyStake(uint256 _qtyToProcess) external onlyRole(DEFAULT_ADMIN_ROLE)  {
         require(airdropStarted, "Airdrop not started");
+        require(!airdropEnded, "Airdrop already complete");
+        require(stakeCounter <= totalSupplyAtStart, "Airdrop complete");
 
-        for (uint256 j = 0; j < keyIds.length; j++) {
-            uint256 keyId = keyIds[j];
-            uint256 startKeyId = keyToStartEnd[keyId][0];
-            uint256 endKeyId = keyToStartEnd[keyId][1];
-
-            require(endKeyId > startKeyId, "Invalid input");
-
-            Referee9 referee = Referee9(refereeAddress);
-            address owner = NodeLicense8(nodeLicenseAddress).ownerOf(keyId);
+        // Start where we left off
+        uint256 startingKeyId = stakeCounter;
 
-            address poolAddress = referee.assignedKeyToPool(keyId);
+        // Ensure we don't go over the total supply
+        uint256 endingKeyId = Math.min(stakeCounter + _qtyToProcess, totalSupplyAtStart);
 
-            //TODO do we need to allow this and just return ?
-            // require(poolAddress != address(0), "Key not staked");
-            if(poolAddress == address(0)){
-                continue;
-            }
+        // Connect to the referee and node license contracts
+        NodeLicense8 nodeLicense = NodeLicense8(nodeLicenseAddress);
+        Referee9 referee = Referee9(refereeAddress);
 
-            //Array size needs to be 1 more than the difference between the start and end key ids
-            uint256[] memory stakeKeyIds = new uint256[](endKeyId - startKeyId + 1);
-            for (uint256 i = 0; i < stakeKeyIds.length; i++) {
-                stakeKeyIds[i] = startKeyId + i;
+        // Loop through the range of node licenses
+        // Needs to be <= to include the last key
+        for (uint256 i = startingKeyId; i <= endingKeyId; i++) {
+            // Get the owner of the node license
+            address owner = nodeLicense.ownerOf(i);
+
+            // Get the pool address for the owner
+            address poolAddress = referee.assignedKeyToPool(i);
+
+            // If the pool address is not 0, stake the newly minted keys
+            if(poolAddress != address(0)){                
+                uint256[] memory stakeKeyIds = new uint256[](keyMultiplier);
+                // Determine the initial token id for the newly minted keys
+                // Calculate the starting ID for the new batch of tokens
+                uint256 tokensAlreadyMinted = (i - 1) * keyMultiplier;
```

Comment:
Could be renamed to tokensAlreadyProcessed, as we are not minting here just staking ?